File: Programmers/level 2/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let workDay=[];
    let releaseDay=0;
    workDay=progresses.map((v,i)=>Math.ceil((100-v)/speeds[i])); //작업이 소요되는 날짜
    //console.log(workDay)
    for (let i=0; i<workDay.length; i++){
        if (releaseDay<workDay[i]){
            answer.push(1);
            releaseDay=workDay[i]; 
        } //배포 날에 준비가 되지 않은 경우
        else{
            answer[answer.length-1]++; //배포 날에 준비가 된 경우
        }
    }
    return answer;
}

File: Programmers/level 2/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> release;
    int n = progresses.size();
    for (int i=0; i<n; i++) {
        release.push_back(ceil(double(100-progresses[i])/speeds[i]));
    }
    
    int day=0;
    
    for (int i=0; i<n; i++) {
        if (release[i]>day) {
            answer.push_back(1);
            day=release[i];
        }
        else {
            answer.back()++;
        }
    }

    return answer;
}
